web/src/echo.ts

```diff
@@ -367,14 +460,23 @@ export function update_message_lists({old_id, new_id}) {
     }
 }
 
-export function process_from_server(messages) {
+export function process_from_server(messages: ServerMessage[]): ServerMessage[] {
```

Comment:
Let's finally write a zod schema to parse the incoming message object from the server here. Ideally, we should parse it as soon as we get it from our events API which happens in `server_events.js` inside `get_events_success` method.
After writing the schema we can then infer the `ServerMessage` type from it.

Reply:
I agree. Added as prep commit 

- https://github.com/zulip/zulip/pull/30702/commits/d82fadb7f6949ffc512951a0d93dfb40818e055b